test/wpt/status/streams.json

```diff
@@ -6,6 +6,10 @@
     "skip": "Browser-specific test"
   },
   "readable-byte-streams/bad-buffers-and-views.any.js": {
-    "fail": "TODO: implement detached ArrayBuffer support"
```

Comment:
hmm, I am not sure how this is an improvement - the original description is a TODO describing why the tests are failing, the new ones are simply copying the test descriptions in WPT, and it has nothing to do with the implementation status of Node.js. This is also the case with several `fail` schema changes below. Updating them to `skip` is a bit pointless too -- they are `fail` because once someone implement these, the test would fail because the expected failures are gone, so the implementers would know that they should update the test expectations. If they are `skip` then the implementer could totally forget about the expectations and the tests would be skipped forever until someone remembers to update the status file to match the state of the implementation.

Reply:
Please see the PR description, how you describe the runner works does not reflect the reality.

WPTs are often thousands of tests and one general fail description doesn't help at all. Having an enumeration of all unique failing test names otoh does allow to identify both expected and unexpected failures as well as unexpected passes.

Reply:
See eg. https://github.com/nodejs/node/pull/44173/commits/b47b2efe3a6a9515df15100bd64f73e34ac9b50c which builds on top of this change. Fixes now need changes to the wpt status.

Reply:
This is still a regression for tests that don't use a .json fail like the WebCrypto tests though (for example "readable-byte-streams/bad-buffers-and-views.any.js", and some below). Why can't we make the schema of `fail` polymorphic for both types of failures?

Reply:
How is it a regression? Please elaborate.

Reply:
Running the tests and ignoring the result is the same as not running it all.

Reply:
Another cleaner solution may be to invent a new field for failures like this, and keep `fail` for simpler tests that don't run import tons of tests from a JSON.

Reply:
> How is it a regression? Please elaborate.

Because we are losing information about why the tests are failing (I was talking about a regression of the readability of the status file). These messages are not arbitrary, they are the result of previous investigations into those failures. `TODO: implement detached ArrayBuffer support` gives you more information about how the failure can be fixed, compared to `ReadableStream with byte source: respondWithNewView() throws if the supplied view's buffer is zero-length (in the readable state)`, for example.

Another solution that might be better in terms of not losing the information would be to make the `expected` a key-value map, where the key is the test description and the value is the reason.

Reply:
Yes, I think the information about why the tests are failing (or why they can't be / are still not fixed, specifically) is valuable and should not be removed.

Reply:
@joyeecheung please review, there are a number of fixes lined up and ready to go dependant on this work.